### ISSUE TITLE
gettilepos（）

### ISSUE THREAD

**Issue by rokkiemcPlayer** [t04glovern/pyCraft#1]
i cant get the“getTilePos” runing，programm get stuck every time at mc.player.getTilePos（），do u know some reason for this？thx

**Comment by t04glovern** [comment]
Is it throwing any particular error? The correct usage for getTilePos() is as follows

`from mcpi.minecraft import Minecraft`
`mc = Minecraft.create()`

`# get player position and put it in a variable called pos`
`pos = mc.player.getTilePos()`
`# get the vector values for x, y and z and place them into their own variables`
`x = pos.x`
`y = pos.y`
`z = pos.z`

Things you'll need to watch for is to make sure your 'mc' matches the 'mc' created in the 'mc = Minecraft.create()' statement

**Comment by rokkiemcPlayer** [comment]
theres no error throwing，i add a line postToChat（"hi"）,when add it on the top of the gettilepos,i can get it from the game,but besides it ,there's no hi and the shell show nothing

**Comment by t04glovern** [comment]
Yeah, sounds normal. postToChat() will only show an output to the minecraft console. What happens if you try something like the following:

```
from mcpi.minecraft import Minecraft
mc = Minecraft.create()

pos = mc.player.getTilePos()
x = pos.x
y = pos.y
z = pos.z

mc.postToChat("X: " + str(x))
mc.postToChat("Y: " + str(y))
mc.postToChat("Z: " + str(z))
```

It should output your players vector positioning to the Minecraft game chat. If that works then you know at least the function is working

**Comment by rokkiemcPlayer** [comment]
Thx for patience!
I'll let you know when I get something new

**Comment by rokkiemcPlayer** [comment]
still nothing
this is the program

**Comment by rokkiemcPlayer** [comment]
still nothing
**this is the program**

from mcpi.minecraft import Minecraft
mc = Minecraft.create()
pos = mc.player.getTilePos()
x = pos.x
y = pos.y
z = pos.z
print(pos.x)
print(pos.y)
print(pos.z)

 **this is the bukkit**

[RaspberryJuice] Task #2 for RaspberryJuice v1.8 generated an exception
java.lang.NoSuchMethodError: org.bukkit.Bukkit.getOnlinePlayers()Ljava/util/Collection;at net.zhuoweizhang.raspberryjuice.RaspberryJuicePlugin.getNamedPlayer(RaspberryJuicePlugin.java:91) ~[?:?]
    at net.zhuoweizhang.raspberryjuice.RemoteSession.getCurrentPlayer(RemoteSession.java:473) ~[?:?]
    at net.zhuoweizhang.raspberryjuice.RemoteSession.handleCommand(RemoteSession.java:238) ~[?:?]
    at net.zhuoweizhang.raspberryjuice.RemoteSession.handleLine(RemoteSession.java:119) ~[?:?]
    at net.zhuoweizhang.raspberryjuice.RemoteSession.tick(RemoteSession.java:99) ~[?:?]
    at net.zhuoweizhang.raspberryjuice.RaspberryJuicePlugin$TickHandler.run(RaspberryJuicePlugin.java:155) ~[?:?]
    at org.bukkit.craftbukkit.v1_7_R2.scheduler.CraftTask.run(CraftTask.java:53) ~[craftbukkit.jar:git-Bukkit-1.7.2-R0.3-14-g8f8716c-b3042jnks]
    at org.bukkit.craftbukkit.v1_7_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:345) [craftbukkit.jar:git-Bukkit-1.7.2-R0.3-14-g8f8716c-b3042jnks]
    at net.minecraft.server.v1_7_R2.MinecraftServer.v(MinecraftServer.java:590) [craftbukkit.jar:git-Bukkit-1.7.2-R0.3-14-g8f8716c-b3042jnks]
    at net.minecraft.server.v1_7_R2.DedicatedServer.v(DedicatedServer.java:250) [craftbukkit.jar:git-Bukkit-1.7.2-R0.3-14-g8f8716c-b3042jnks]
    at net.minecraft.server.v1_7_R2.MinecraftServer.u(MinecraftServer.java:548) [craftbukkit.jar:git-Bukkit-1.7.2-R0.3-14-g8f8716c-b3042jnks]
    at net.minecraft.server.v1_7_R2.MinecraftServer.run(MinecraftServer.java:459) [craftbukkit.jar:git-Bukkit-1.7.2-R0.3-14-g8f8716c-b3042jnks]
    at net.minecraft.server.v1_7_R2.ThreadServerApplication.run(SourceFile:618) [craftbukkit.jar:git-Bukkit-1.7.2-R0.3-14-g8f8716c-b3042jnks]
[RaspberryJuice] Closed connection to/127.0.0.1:49215.
[RaspberryJuice] Starting output thread!
[RaspberryJuice] Starting input thread
 [RaspberryJuice] Opened connection to/127.0.0.1:49217.
 [RaspberryJuice] Task #2 for RaspberryJuice v1.8 generated an exception
java.lang.NoSuchMethodError: org.bukkit.Bukkit.getOnlinePlayers()Ljava/util/Collection;
    at net.zhuoweizhang.raspberryjuice.RaspberryJuicePlugin.getNamedPlayer(RaspberryJuicePlugin.java:91) ~[?:?]
    at net.zhuoweizhang.raspberryjuice.RemoteSession.getCurrentPlayer(RemoteSession.java:473) ~[?:?]
    at net.zhuoweizhang.raspberryjuice.RemoteSession.handleCommand(RemoteSession.java:238) ~[?:?]
    at net.zhuoweizhang.raspberryjuice.RemoteSession.handleLine(RemoteSession.java:119) ~[?:?]
    at net.zhuoweizhang.raspberryjuice.RemoteSession.tick(RemoteSession.java:99) ~[?:?]
    at net.zhuoweizhang.raspberryjuice.RaspberryJuicePlugin$TickHandler.run(RaspberryJuicePlugin.java:155) ~[?:?]
    at org.bukkit.craftbukkit.v1_7_R2.scheduler.CraftTask.run(CraftTask.java:53) ~[craftbukkit.jar:git-Bukkit-1.7.2-R0.3-14-g8f8716c-b3042jnks]
    at org.bukkit.craftbukkit.v1_7_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:345) [craftbukkit.jar:git-Bukkit-1.7.2-R0.3-14-g8f8716c-b3042jnks]
    at net.minecraft.server.v1_7_R2.MinecraftServer.v(MinecraftServer.java:590) [craftbukkit.jar:git-Bukkit-1.7.2-R0.3-14-g8f8716c-b3042jnks]
    at net.minecraft.server.v1_7_R2.DedicatedServer.v(DedicatedServer.java:250) [craftbukkit.jar:git-Bukkit-1.7.2-R0.3-14-g8f8716c-b3042jnks]
    at net.minecraft.server.v1_7_R2.MinecraftServer.u(MinecraftServer.java:548) [craftbukkit.jar:git-Bukkit-1.7.2-R0.3-14-g8f8716c-b3042jnks]
    at net.minecraft.server.v1_7_R2.MinecraftServer.run(MinecraftServer.java:459) [craftbukkit.jar:git-Bukkit-1.7.2-R0.3-14-g8f8716c-b3042jnks]
    at net.minecraft.server.v1_7_R2.ThreadServerApplication.run(SourceFile:618) [craftbukkit.jar:git-Bukkit-1.7.2-R0.3-14-g8f8716c-b3042jnks]
 [RaspberryJuice] Closed connection to/127.0.0.1:49345.
 [RaspberryJuice] Starting output thread!

**Comment by t04glovern** [comment]
Keep in mind that when you run the print() function you'll only get an output to the python interpreter command line (i.e. you won't see any output in the bukkit server logs)

I just tried your code line for line and it seems to work fine

**Code**

```
from mcpi.minecraft import Minecraft
mc = Minecraft.create()
pos = mc.player.getTilePos()
x = pos.x
y = pos.y
z = pos.z
print(pos.x)
print(pos.y)
print(pos.z)
```

**Output**

```
Python 3.5.1 (v3.5.1:37a07cee5969, Dec  6 2015, 01:54:25) [MSC v.1900 64 bit (AMD64)] on win32
Type "copyright", "credits" or "license()" for more information.
>>> 
================= RESTART: C:/Users/Nathan/Downloads/test.py =================
97
73
249
>>> 
```

I am running a pre-build server package with all the python api's prebuld in (https://sourceforge.net/projects/program-with-minecraft/).

**Minecraft Server Version**
`[10:40:33 INFO]: Starting minecraft server version 1.9.4`

**Bukkit Version**
`[10:40:33 INFO]: This server is running CraftBukkit version git-Spigot-798f32d-0cd0397 (MC: 1.9.4) (Implementing API version 1.9.4-R0.1-SNAPSHOT)`

**RaspberryJuice Version**
`[10:40:41 INFO]: [RaspberryJuice] Enabling RaspberryJuice v1.7`

My only recommendation is to make sure you're running a supported version of python (3.5+) and maybe confirm that your version of RaspberryJuice doesn't have any known errors.

If that doesn't solve your problem, maybe try the server packages I used while making this repository (https://sourceforge.net/projects/program-with-minecraft/)
